packages/serverpod/lib/src/server/session.dart

```diff
@@ -146,24 +142,28 @@ abstract class Session {
     }
 
     if (enableLogging) {
-      var logWriter = _createLogWriter(
-        this,
-        server.serverpod.logSettingsManager,
-      );
-      _logManager = SessionLogManager(
-        logWriter,
-        session: this,
-        settingsForSession: (Session session) => server
-            .serverpod.logSettingsManager
-            .getLogSettingsForSession(session),
-        disableLoggingSlowSessions: _isLongLived(this),
-        serverId: server.serverId,
-      );
-    } else {
-      _logManager = null;
+      startLogging();
     }
   }
 
+  /// Turns on logging for the session if it was not already turned on.
+  void startLogging() {
```

Comment:
Suggestion: Make private we should not expose this to the users.

Reply:
maybe move back into constructor? question pending further up though.

Reply:
Reverted here: [cec5fad](https://github.com/serverpod/serverpod/pull/2676/commits/cec5fad3d1dececc42a205ca6662da6c4342f464)